web/src/test/java/com/microsoft/applicationinsights/web/internal/WebRequestTrackingFilterTests.java

```diff
@@ -156,6 +146,18 @@ public void testUnhandledIOExceptionWithTelemetryClientThatThrows() throws Illeg
         testException(createdData, new IOException());
     }
 
+    @Test
+    public void testWebAppNameInitializerAppNameHandoff() throws Exception {
+        FilterAndTelemetryClientMock mocks = createInitializedFilterWithExposedTelemetryClient();
+        FilterChain chain = mock(FilterChain.class);
+
+        mocks.filter.doFilter(ServletUtils.generateDummyServletRequest(), ServletUtils.generateDummyServletResponse(), chain);
+        String role = mocks.mockTelemetryClient.getContext().getCloud().getRole();
+        assertNotNull(role);
+        assertEquals(this.mockContextPath.replace("/", ""), role);
+
+    }
+
     // region Private methods
```

Comment:
This kind of C# stuff also we should consider cleaning :)

Reply:
`// region` is recognized by IntelliJ.